### PR TITLE
terraform-provider-lxd: init at 2019-04-03

### PR DESCRIPTION
**Pull request by blogle**
###### Motivation for this change
Adds terraform provider for provisioning images on the lxc "hypervisor". This was tested on NixOS by enabling `virtualization.lxd` in `/etc/nixos/configuration.nix` and finally provisioning an ubuntu image on the hypervisor via terraform.
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2759,7 +2759,6 @@
     name = "Linquize";
   };
   linus = {
-    email = "linusarver@gmail.com";
```
Comment (c0bw3b) [inline maintainers/maintainer-list.nix]: Oops? Please keep this line

File: pkgs/applications/networking/cluster/terraform-providers/lxd/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# This file was generated by https://github.com/kamilchm/go2nix v1.3.0
+{ stdenv, buildGoPackage, fetchgit }:
+
+buildGoPackage rec {
+  name = "terraform-provider-lxd-unstable-${version}";
+  version = "2019-04-03";
```
Comment (c0bw3b) [inline pkgs/applications/networking/cluster/terraform-providers/lxd/default.nix]: Why not the stable v1.1.3 ?

File: pkgs/applications/networking/cluster/terraform-providers/lxd/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  rev = "ed5c4247171d35f012c765c585cd24a44e4f234e";
+
+  goPackagePath = "github.com/sl1pm4t/terraform-provider-lxd";
+
+  src = fetchgit {
```
Comment (c0bw3b) [inline pkgs/applications/networking/cluster/terraform-providers/lxd/default.nix]: You can use `fetchFromGitHub` instead

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (blogle) [reply maintainers/maintainer-list.nix]: wow, oops not sure how this occurred.

File: pkgs/applications/networking/cluster/terraform-providers/lxd/default.nix (hunk 2 above)
Comment (blogle) [reply pkgs/applications/networking/cluster/terraform-providers/lxd/default.nix]: v1.1.3 would not build for me citing `go/src/github.com/sl1pm4t/terraform-provider-lxd/lxd/resource_lxd_snapshot.go:128:29: snap.CreationDate undefined (type *api.ContainerSnapshot has no field or method CreationDate)`

File: pkgs/applications/networking/cluster/terraform-providers/lxd/default.nix (hunk 3 above)
Comment (blogle) [reply pkgs/applications/networking/cluster/terraform-providers/lxd/default.nix]: This derivation was auto generated with go2nix. I thought it better not to modify auto generated code

**Comment by blogle** [comment]
bump. can I get a merge?

**Pull request closed without merging**